### PR TITLE
Implemented new way of managing dependencies and updated them

### LINKED ISSUE #4691: Investigate the best way to manage dependency versions across modules
Right now we have a bunch of dependencies shared among modules and have been using version constants to keep them in sync where we can. Is there a better way to do this? 

If not, we should sweep the modules we have so far and try and pull out as many constants as we can.

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #4691

#### What has been done to verify that this works as intended?
I ran automated tests.

#### Why is this the best possible solution? Were any other approaches considered?
We considered options described in https://proandroiddev.com/how-to-manage-dependencies-in-multi-module-project-84620afbb415 and decided that the one with buildSrc module is the best one.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
It doesn't require testing. As always we assume that dependency updates will be tested during our every day work.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: config/pmd-ruleset.xml [hunk 1]
```diff
@@ -17,6 +17,7 @@
         <exclude name="AvoidReassigningParameters" />
         <exclude name="AvoidStringBufferField" />
         <exclude name="ForLoopCanBeForeach" />
+        <exclude name="GuardLogStatement" />
```
Comment (grzesiek2010) [inline config/pmd-ruleset.xml:20]: I will review those new pmd rules and enable later. The same with checkstyle.

File: analytics/build.gradle [hunk 2]
```diff
@@ -32,9 +35,8 @@ android {
 }
 
 dependencies {
-
-    implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
-    implementation "androidx.core:core-ktx:${rootProject.coreKtxVersion}"
-    implementation 'com.google.firebase:firebase-analytics:18.0.1'
-    implementation 'com.google.firebase:firebase-crashlytics:17.3.0'
+    implementation Dependencies.kotlin_stdlib
```
Comment (seadowg) [inline analytics/build.gradle:38]: Wow this is so much nicer!

Comment (seadowg) [inline analytics/build.gradle:38]: Wow this is so much nicer!

File: buildSrc/src/main/java/dependencies/Dependencies.kt [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    const val kotlin_stdlib_jdk7 = "org.jetbrains.kotlin:kotlin-stdlib-jdk7:${Versions.kotlin}"
+    const val json = "org.json:json:20210307"
+    const val gson = "com.google.code.gson:gson:2.8.7"
+    const val firebase_analytics = "com.google.firebase:firebase-analytics:19.0.0"
+    const val firebase_crashlytics = "com.google.firebase:firebase-crashlytics:18.2.1"
+    const val findbugs_annotations = "com.google.code.findbugs:annotations:3.0.1"
+    const val findbugs_jsr305 = "com.google.code.findbugs:jsr305:3.0.2"
+    const val fastlane_screengrab = "tools.fastlane:screengrab:2.1.0"
+    const val leakcanary = "com.squareup.leakcanary:leakcanary-android:2.7"
+    const val timber = "com.jakewharton.timber:timber:${Versions.timber}"
```
Comment (seadowg) [inline buildSrc/src/main/java/dependencies/Dependencies.kt]: I don't think Timber needs its own `Version` constant as it's only used once? Also, Timber was just updated to fix a bug we ran into (https://github.com/JakeWharton/timber/issues/434#event-5155819740). Lets update here and remove the ignore in `lint.xml` on `TimberArgCount`.

File: config/pmd-ruleset.xml (hunk 1 above)
Comment (seadowg) [reply config/pmd-ruleset.xml:20]: Can we create an issue for that?

**Review by seadowg — CHANGES_REQUESTED** [review]
This is so nice! One small change to make while we're here, I think.

File: config/pmd-ruleset.xml (hunk 1 above)
Comment (grzesiek2010) [reply config/pmd-ruleset.xml:20]: https://github.com/getodk/collect/issues/4794

File: buildSrc/src/main/java/dependencies/Dependencies.kt (hunk 3 above)
Comment (grzesiek2010) [reply buildSrc/src/main/java/dependencies/Dependencies.kt]: Done.

**Review by seadowg — APPROVED** [review]

**Pull request merged**